### PR TITLE
[Bug] Fix Arrow-FS parquet reader for larger files

### PR DESCRIPTION
**Pull request by rjzamora**
## Description
Follow-up to https://github.com/rapidsai/cudf/pull/16684

There is currently a bug in `dask_cudf.read_parquet(..., filesystem="arrow")` when the files are larger than the `"dataframe.parquet.minimum-partition-size"` config. More specifically, when the files are not aggregated together, the output will be `pd.DataFrame` instead of `cudf.DataFrame`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/dask_cudf/dask_cudf/io/tests/test_parquet.py [hunk 1]
```diff
@@ -615,3 +619,24 @@ def test_timezone_column(tmpdir):
     got = dask_cudf.read_parquet(path)
     expect = cudf.read_parquet(path)
     dd.assert_eq(got, expect)
+
+
+@require_dask_expr()
+@pytest.mark.parametrize("min_part_size", ["1B", "1GB"])
+def test_read_parquet_arrow_filesystem(tmpdir, min_part_size):
```
Comment (rjzamora) [inline python/dask_cudf/dask_cudf/io/tests/test_parquet.py:626]: TODO: Skip if `not dask_cudf.backends.PYARROW_GE_15`

File: python/dask_cudf/dask_cudf/expr/_expr.py [hunk 2]
```diff
@@ -99,6 +99,31 @@ def _table_to_pandas(
...
+                self._fragment_to_table,
+                FragmentWrapper(self.fragments[index], filesystem=self.fs),
+                self.filters,
+                columns,
+                schema,
+            ),
+            index_name,
+            None,
+            None,
+            None,
```
Comment (madsbk) [inline python/dask_cudf/dask_cudf/expr/_expr.py:124]:
```suggestion
```
Do we need this?

**Review by madsbk — APPROVED** [review]
Looks good to me, thanks @rjzamora

**Pull request merged**